### PR TITLE
[MAINTENANCE] Add VersionedLink component

### PR DESCRIPTION
**Pull request by tyler-hoffman**
Adds the tooling to create version safe links. Comments show the use cases. Docusaurus will throw during `yarn build` if we build any bad urls with this!

- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: docs/docusaurus/src/components/VersionedLink/index.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+/**
+ * Version-safe link component.
+ * 
+ * The `to` prop MUST be prefixed with a slash and starting from the content root.
+ * e.g. <VersionedLink to="/intro" /> used in a page at `/docs/1.0.0/foo/bar` links to `docs/1.0.0/intro
+ * e.g. <VersionedLink to="/intro" /> used in a page at `/docs/foo/bar` links to `docs/intro
+ */
+const VersionedLink = ({to, ...rest}) => {
+    const versionedPath = useVersionedPath(to);
```
Comment (billdirks) [inline docs/docusaurus/src/components/VersionedLink/index.js:13]: Do you want the validate that `to` starts with a `/`?

Oh, I see you do this in the other file.

File: docs/docusaurus/src/hooks/use_versioned_path.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+export const useVersionedPath = (path) => {
+    console.log({path})
+    if (!path.startsWith('/')) {
+        throw Error("paths to useVersionedLink must start with a slash")
+    }
+    const {pathname} = useLocation();
+    const {versions} = usePluginData('docusaurus-plugin-content-docs');
+    for (const version of Object.values(versions)) {
+        if (version.path === '/docs') {
+            // Skip the "current" version, since it matches all acses
```
Comment (billdirks) [inline docs/docusaurus/src/hooks/use_versioned_path.js]: Should this be s/acses/cases/

**Review by billdirks — APPROVED** [review]
LGTM.

File: docs/docusaurus/src/hooks/use_versioned_path.js (hunk 2 above)
Comment (tyler-hoffman) [reply docs/docusaurus/src/hooks/use_versioned_path.js]: Nope, `acses` is super advanced documentation lingo 😛

**Pull request merged**